opensrp-fp/src/main/java/org/smartregister/fp/common/task/FinalizeVisitFormTask.java

```diff
@@ -0,0 +1,174 @@
+package org.smartregister.fp.common.task;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.text.TextUtils;
+
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+import com.vijay.jsonwizard.rules.RuleConstant;
+
+import org.joda.time.LocalDate;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.smartregister.fp.R;
+import org.smartregister.fp.common.domain.Contact;
+import org.smartregister.fp.common.domain.WomanDetail;
+import org.smartregister.fp.common.library.FPLibrary;
+import org.smartregister.fp.common.repository.PreviousContactRepository;
+import org.smartregister.fp.common.rule.ContactRule;
+import org.smartregister.fp.common.util.ConstantsUtils;
+import org.smartregister.fp.common.util.DBConstantsUtils;
+import org.smartregister.fp.common.util.FPFormUtils;
+import org.smartregister.fp.common.util.Utils;
+import org.smartregister.fp.features.home.repository.PatientRepository;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import timber.log.Timber;
+
+import static org.smartregister.fp.common.util.Utils.isCheckboxValueEmpty;
+
+public class FinalizeVisitFormTask extends AsyncTask<Void, Void, Void> {
+
+    private final Context context;
+    private final String baseEntityId;
+    private final int contactNo;
+    private final Contact contact;
+    private final String jsonCurrentState;
+    private final FPFormUtils fpFormUtils;
+    private final ProgressDialog progressDialog;
+    private HashMap<String, String> clientProfileDetail;
+
+    public FinalizeVisitFormTask(Context context, String baseEntityId, int contactNo, Contact contact, String jsonCurrentState) {
+        this.context = context;
+        this.baseEntityId = baseEntityId;
+        this.contactNo = contactNo;
+        this.contact = contact;
+        this.jsonCurrentState = jsonCurrentState;
+        fpFormUtils = new FPFormUtils();
+
+        progressDialog = new ProgressDialog(context);
+        progressDialog.setCancelable(false);
+        progressDialog.setTitle(context.getString(com.vijay.jsonwizard.R.string.loading));
+        progressDialog.setMessage(context.getString(R.string.finalizing_form_message));
+        progressDialog.show();
+    }
+
+    @Override
+    protected Void doInBackground(Void... voids) {
+
+        contact.setJsonForm(fpFormUtils.addFormDetails(jsonCurrentState));
+        contact.setContactNumber(contactNo);
+        FPFormUtils.persistPartial(baseEntityId, contact);
+
+        /*Map<String, String> clientProfileDetail = PatientRepository.getClientProfileDetails(baseEntityId);
+        ContactInteractor contactInteractor = new ContactInteractor();
+        contactInteractor.finalizeContactForm(clientProfileDetail, this);*/
+
+        clientProfileDetail = PatientRepository.getClientProfileDetails(baseEntityId);
+
+        int gestationAge = getGestationAge(clientProfileDetail);
```

Comment:
`clientProfileDetail` is possibly a null object as returned by `PatientRepository.getClientProfielDetails`. This is may result to an NPE

Reply:
added condition for check NPE